lldb/tools/lldb-dap/src-ts/extension.ts

```diff
@@ -17,14 +17,46 @@ function createDefaultLLDBDapOptions(): LLDBDapOptions {
       const path = vscode.workspace
         .getConfiguration("lldb-dap", session.workspaceFolder)
         .get<string>("executable-path");
-      if (path) {
-        return new vscode.DebugAdapterExecutable(path, []);
+
+      if (!path) {
+        return packageJSONExecutable;
       }
-      return packageJSONExecutable;
+
+      vscode.workspace.fs.stat(vscode.Uri.file(path)).then(
+        (fileStats) => {
```

Comment:
in addition to checking the presence of the file on launch, we could also check for it in the `onDidChangeConfiguration` event if the `executable-path` gets changed.

(100% optional; feel free to just ignore this comment 🙂)